client/blocks/subscription-length-picker/index.jsx

```diff
@@ -0,0 +1,193 @@
+import { localize } from 'i18n-calypso';
+import formatCurrency from 'lib/format-currency';
+import { getProductsList, isProductsListFetching } from 'state/products-list/selectors';
+import { requestProductsList } from 'state/products-list/actions';
+import { requestPlans } from 'state/plans/actions';
+import { getPlanRawPrice, isRequestingPlans } from '../../state/plans/selectors';
+import { getCurrentUserCurrencyCode } from '../../state/current-user/selectors';
+import { getPlanDiscountedRawPrice } from '../../state/sites/plans/selectors';
+import { getSelectedSiteId } from '../../state/ui/selectors';
+import { abtest } from 'lib/abtest';
+import { getPlan, applyTestFiltersToPlansList } from 'lib/plans';
+import { TERM_MONTHLY } from 'lib/plans/constants';
+import SubscriptionLengthOption from './option';
+
+export class SubscriptionLengthPicker extends React.Component {
+	static propTypes = {
+		initialValue: PropTypes.string,
+		currencyCode: PropTypes.string.isRequired,
+		productsWithPrices: PropTypes.array.isRequired,
+		onChange: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		onChange: () => null,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			checked: props.initialValue,
+		};
+	}
+
+	render() {
+		const { productsWithPrices, translate } = this.props;
+		return (
+			<div className="subscription-length-picker">
+				<div className="subscription-length-picker__header">
+					{ translate( 'Choose the length of your subscription', {
+						comment:
+							'We offer a way to change billing term from default bill every 1 year to something ' +
+							'else like bill once every 2 years - this is related header.',
+					} ) }
+				</div>
+
+				<div className="subscription-length-picker__options">
+					{ productsWithPrices.map( p => this.renderProduct( p ) ) }
+				</div>
+			</div>
+		);
+	}
+
+	renderProduct = p => {
+		const { plan, planSlug } = p;
+
+		return (
+			<div className="subscription-length-picker__option-container" key={ planSlug }>
+				<SubscriptionLengthOption
+					term={ plan.term }
+					checked={ planSlug === this.state.checked }
+					price={ myFormatCurrency( p.priceFull, this.props.currencyCode ) }
+					pricePerMonth={ myFormatCurrency( p.priceMonthly, this.props.currencyCode ) }
+					savePercent={ Math.round( 100 * ( 1 - p.priceMonthly / this.getHighestMonthlyPrice() ) ) }
+					value={ planSlug }
+					onCheck={ this.handleCheck }
+				/>
+			</div>
+		);
+	};
+
+	getHighestMonthlyPrice() {
+		return Math.max( ...this.props.productsWithPrices.map( p => Number( p.priceMonthly ) ) );
+	}
+
+	handleCheck = ( { value } ) => {
+		this.setState( {
+			checked: value,
+		} );
+		this.props.onChange( { value } );
+	};
+}
+
+class SubscriptionLengthPickerContainer extends React.Component {
+	static defaultProps = {
+		...SubscriptionLengthPicker.defaultProps,
```

Comment:
Please avoid the temptation to do things like this. Let's optimize for readability vs loc. Folks reading out of context will need to skip through multiple files.

Reply:
I do not agree, copying&pasting props in this case will introduce a need to always remember about updating both declarations - and eventually they will get out of sync. That being said, with both components merged this is moot.

Reply:
> in this case will introduce a need to always remember about updating both declarations

It depends on the circumstances, but in some cases this may also be a code smell. We have global state available. If we find that we're passing a lot of props to a very specific child component, maybe we could consider connecting the child should to global state.